SpeechResponder/ScriptResolver.cs

```diff
@@ -1186,9 +1186,13 @@ CommodityMarketQuote CommodityDetails(string commodityLocalizedName, Station sta
 
                 if (curr?.x != null && dest?.x != null)
                 {
-                    result = (decimal)Math.Round(Math.Sqrt(square((double)(curr.x - dest.x))
-                                + square((double)(curr.y ?? 0 - dest.y ?? 0))
-                                + square((double)(curr.z ?? 0 - dest.z ?? 0))), 2);
+                    result = (decimal)Math.Round(
+                        Math.Sqrt(
+                                square((double)(curr.x - dest.x)) +
+                                square((double)((curr.y ?? 0) - (dest.y ?? 0))) +
```

Comment:
I don't like the defaulting to 0 here. These are [null-lifted operators](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#lifted-operators) and will propogate null if either side is null.

Reply:
> It isn't clear to me from the diff where the error was. Please can we write that up?

I believe 
```cs
square((double)(curr.y ?? 0 - dest.y ?? 0))
```
was being interpreted like
```cs
square((double)(curr.y ?? (0 - dest.y) ?? 0))
```
when
```cs
square((double)((curr.y ?? 0) - (dest.y ?? 0)))
```
was intended.